dracut/30ignition/boot.mount

```diff
@@ -0,0 +1,15 @@
+[Unit]
+Description=Ignition Boot Mount
+Documentation=https://github.com/coreos/ignition
+
+# Make sure that the device exists
+Requires=dev-disk-by\x2dlabel-boot.device
+After=dev-disk-by\x2dlabel-boot.device
```

Comment:
I'm not sure we can require `label=boot` here in ignition-dracut; would it work for MicroOS?  Similarly Fedora IoT is using Ignition but not coreos-assembler (but not from the initramfs, so this may not matter).  

I'm not opposed to adding it here but I think because the "label" stuff is coreos-assembler it should likely live in fedora-coreos-config.

Reply:
cc @laenion

Reply:
Based on the dependencies, this shouldn't break 'em. The change means that:
- if `LABEL=boot` is not present, then the unit won't activate
- the unit would have failed already since it _would_ try to do the mount anyway
- `ignition-setup-user.service` is `After`, there is no `Requires`
- The `boot.mount` is only bound to diskful targets

Reply:
`install_ignition_unit` is adding it as a `Requires`, and I think non-ostree users are using `ignition-diskful.target`.

Reply:
Now, today we do ship `ignition-firstboot-complete.service` in this repo which mounts `/boot` - but it runs in the real root currently.

Clearly if we try to move firstboot-complete into the initramfs that would be a forcing-function for this - we'd need to standardize (or make configurable) how to find `/boot`.

Making this configurable is also an option, just a bit ugly.

Reply:
What am I missing? [1] https://github.com/coreos/ignition-dracut/blob/master/dracut/30ignition/ignition-generator#L53-L58 add the device label as a runtime dependency. This change actually gives people more flexibility.

Reply:
Yeah, looks like you're right that unit already has a dependency on the boot label.  So presumably either MicroOS has that label, or they haven't updated ignition-dracut, or they're patching it.

Reply:
> I'm not sure we can require `label=boot` here in ignition-dracut; would it work for MicroOS?

Putting the `/boot` dependency into a separate unit won't make it work _out of the box_ on MicroOS, but it would be easy to just overwrite this one unit instead of parts of the generator. It would even be possible to define custom criteria, for example to probe for a USB flash drive, so this would be an excellent change from my point of view.